### PR TITLE
Add README for OpenSearch

### LINKED ISSUE #15: [Documentation] Documentation of Helm Charts for OpenSearch
It will be great if we can have a well written README describing the different options Helm Charts have. It should contain the following components:
1. Instructions about setting up the local dev environment
2. Instructions about installing the chart of k8s
2. Description about different params in `values.yaml` and how to use them.
2. How to use the charts to do different kind of deployments.

P:S: I would love to contribute on this part 😸

### PR DESCRIPTION
**Pull request by TheAlgo**
Signed-off-by: TheAlgo <dhirajjain0@gmail.com>

### Description
Add installation instructions and configuration details for OpenSearch chart.
 
### Issues Resolved
#15 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by TheAlgo** [comment]
@DandyDeveloper @smlx @peterzhuamazon Have a look at this PR when free. Thanks

File: charts/opensearch/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
+# OpenSearch Helm Chart
+
+This Helm chart installs [OpenSearch](https://github.com/opensearch-project/OpenSearch) with configurable TLS, RBAC and much more configuration. This chart caters a number of different use cases and setups.
```
Comment (peterzhuamazon) [inline charts/opensearch/README.md:3]: and much more configuration**S**

File: charts/opensearch/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+- [Requirements](#requirements)
+- [Installing the chart](#installing)
+- [Uninstalling the chart](#uninstalling)
+- [Configuration](#configuration)
+
+## Requirements
+
+* Kubernetes >= 1.14
+* Helm >= 2.17.0
+* Recommended that you have at least 8 GiB of memory available for this deployment. You can expect the deployment to fail if, say, you have less than 4 GiB of memory available.
```
Comment (peterzhuamazon) [inline charts/opensearch/README.md:14]: We recommend you to have 8 GiB of memory available for this deployment, or at least 4 GiB for the minimum requirement. Else, the deployment is expected to fail.

File: charts/opensearch/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+## Installing
+
+To install the chart with the release name `my-release`:
+- Switch to opensearch directly after cloning the repo
+  `❯ cd charts/opensearch`
+- Run `❯ helm package .`
+- Install using Helm 3:
+`❯ helm install my-release opensearch-1.0.0.tgz`
+-  Install using Helm 2
+`❯ helm install --name my-release opensearch-1.0.0.tgz`
```
Comment (peterzhuamazon) [inline charts/opensearch/README.md:25]: Do we still need these if #46 get merged?

File: charts/opensearch/README.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+
+The command deploys OpenSearch with its associated components (data statefulsets, masters, clients) on the Kubernetes cluster in the default configuration.
+
+## Uninstalling
+To delete/uninstall the chart with the release name `my-release`:
```
Comment (peterzhuamazon) [inline charts/opensearch/README.md:31]: Recommend adding `helm list` to check whether the deployment exist before run `helm delete` or `helm uninstall`.

Also, helm delete has been renamed to uninstall: https://helm.sh/docs/helm/helm_delete/

File: charts/opensearch/README.md (hunk 3 above)
Comment (TheAlgo) [reply charts/opensearch/README.md:25]: It depends how people want to install. I will any day prefer the hosted way though.

File: charts/opensearch/README.md (hunk 4 above)
Comment (TheAlgo) [reply charts/opensearch/README.md:31]: Sounds good. Will make the changes

**Comment by TheAlgo** [comment]
@DandyDeveloper Have a look when free. Thanks

File: charts/opensearch/README.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+* Recommended that you have at least 8 GiB of memory available for this deployment. You can expect the deployment to fail if, say, you have less than 4 GiB of memory available.
+
+## Installing
+
+To install the chart with the release name `my-release`:
+- Switch to opensearch directly after cloning the repo
+  `❯ cd charts/opensearch`
+- Run `❯ helm package .`
+- Install using Helm 3:
+`❯ helm install my-release opensearch-1.0.0.tgz`
```
Comment (DandyDeveloper) [inline charts/opensearch/README.md:23]: Why do we ask them to package if they are in the chart dir anyway? They can just `helm install opensearch . ` if they are in the repo.

**Review by DandyDeveloper — APPROVED** [review]
LGTM, one comment, nothing crazy

File: charts/opensearch/README.md (hunk 5 above)
Comment (TheAlgo) [reply charts/opensearch/README.md:23]: Yes they can. I just wanted opinions, as in ODFE this was the doc. I also don't prefer packaging but nevertheless we will move away from it after we have the releasing in place. They users can directly add the repo and install the charts from the hosted repository.

**Comment by TheAlgo** [comment]
@peterzhuamazon I have addressed your comments. Have a look when free. Thanks

**Review by peterzhuamazon — APPROVED** [review]

**Pull request merged**